src/libc/pthread/mutex.rs

```diff
@@ -112,23 +93,22 @@ fn pthread_mutex_init(
     let type_ = if !attr.is_null() {
         check_magic!(env, attr, MAGIC_MUTEXATTR);
         let pthread_mutexattr_t { type_, .. } = env.mem.read(attr);
-        assert!(
-            type_ == PTHREAD_MUTEX_NORMAL
-                || type_ == PTHREAD_MUTEX_ERRORCHECK
-                || type_ == PTHREAD_MUTEX_RECURSIVE
-        );
+        let type_ = type_.try_into().unwrap();
```

Comment:
This does nothing because the `type_` field of the `pthread_mutexattr_t` is a `MutexType`. Since `MutexType` is now an enum rather than an `i32`, so you've accidentally introduced undefined behaviour here.

I caught this because Clippy warned about the useless `try_into()`. Did you maybe forget to run `cargo clippy` and `cargo fmt` before committing? (I should add these to CI…)

Reply:
Thanks for catching this :D

> I caught this because Clippy warned about the useless try_into(). Did you maybe forget to run cargo clippy and cargo fmt before committing? (I should add these to CI…)

Apparently I didn't. 

It might nice to have a derive that checks members for `SafeRead/Write`ability to make this kind of mistake harder. I might get around to it (or might not).